src/overlays/actors/ovl_En_Mkk/z_en_mkk.c

```diff
@@ -82,62 +111,498 @@ static DamageTable D_80A4F754 = {
+}
+
+void func_80A4E0CC(EnMkk* this) {
+    this->alpha = 0;
+    this->unk_14B |= 3;
+    this->actor.flags &= ~ACTOR_FLAG_1;
+    this->actionFunc = func_80A4E100;
+}
+
+void func_80A4E100(EnMkk* this, PlayState* play) {
+    s32 newAlpha;
+
+    if ((this->unk_14B & 4) != 0) {
+        newAlpha = this->alpha + 15;
+    } else {
+        newAlpha = this->alpha + 5;
+    }
+    if (newAlpha >= 255) {
+        this->primColorSelect = 3;
+        this->collider.base.acFlags |= AC_ON;
+        this->alpha = 255;
+        this->actor.flags |= ACTOR_FLAG_1;
+        this->actor.shape.rot.y = this->actor.yawTowardsPlayer;
+        this->unk_14B &= ~4;
+        func_80A4E190(this);
+    } else {
+        this->alpha = newAlpha;
+    }
+}
+
+void func_80A4E190(EnMkk* this) {
+    this->unk_14E = (s32)(Rand_ZeroOne() * 20.0f) + 10;
+    if (this->unk_149 != 0) {
+        this->unk_149--;
+    }
+    this->actionFunc = func_80A4E1F0;
+}
+
+void func_80A4E1F0(EnMkk* this, PlayState* play) {
+    this->unk_14E--;
+    if (this->unk_14E == 0) {
+        func_80A4E2B8(this);
+    }
+}
+
+void func_80A4E22C(EnMkk* this, PlayState* play) {
+    Vec3f pos;
+
+    pos.x = this->actor.world.pos.x;
+    pos.y = this->actor.world.pos.y + 15.0f;
+    pos.z = this->actor.world.pos.z;
+    func_800B0DE0(play, &pos, &gZeroVec3f, &D_80A4F788, &D_80A4F780, &D_80A4F784, 350, 20);
+}
+
+void func_80A4E2B8(EnMkk* this) {
+    this->unk_14E = 30;
+    this->unk_150 = this->actor.shape.rot.y;
+    this->unk_14B &= ~1;
+    this->actionFunc = func_80A4E2E8;
+}
+
+void func_80A4E2E8(EnMkk* this, PlayState* play) {
+    Player* sp24 = GET_PLAYER(play);
+    s32 sp20;
+
+    this->unk_14E--;
+    if ((this->actor.params == 1) && (this->actor.bgCheckFlags & 1) && (this->actor.speedXZ > 2.5f) &&
+        (!(play->gameplayFrames % 3))) {
+        func_80A4E22C(this, play);
+    }
+    if (this->unk_14E > 0) {
+        Math_StepToF(&this->actor.speedXZ, 5.0f, 0.7f);
+        sp20 = 0;
+    } else {
+        sp20 = Math_StepToF(&this->actor.speedXZ, 0.0f, 0.7f);
+    }
+    if ((sp24->stateFlags3 & 0x100) || (Player_GetMask(play) == PLAYER_MASK_STONE)) {
+        Math_ScaledStepToS(&this->unk_150, Actor_YawToPoint(&this->actor, &this->actor.home.pos), 0x400);
+    } else if ((sp24->stateFlags2 & 0x80) || (sp24->actor.freezeTimer > 0)) {
+        Math_ScaledStepToS(&this->unk_150, this->actor.yawTowardsPlayer + 0x8000, 0x400);
+    } else {
+        Math_ScaledStepToS(&this->unk_150, this->actor.yawTowardsPlayer, 0x400);
+    }
+    this->actor.shape.rot.y =
+        (s32)(sin_rad(this->unk_14E * 0.41887903f) * (614.4f * this->actor.speedXZ)) + this->unk_150;
+    func_800B9010(&this->actor, NA_SE_EN_KUROSUKE_MOVE - SFX_FLAG);
+    if (sp20 != 0) {
+        this->unk_14B &= ~2;
+        func_80A4E190(this);
+    } else {
+        if ((this->unk_149 == 0) && (!(sp24->stateFlags3 & 0x100)) && (Player_GetMask(play) != PLAYER_MASK_STONE) &&
```

Comment:
```suggestion
    } else if ((this->unk_149 == 0) && (!(sp24->stateFlags3 & 0x100)) && (Player_GetMask(play) != PLAYER_MASK_STONE) &&
```
This can be a single `else if`